### PR TITLE
feat(storage): support different snapshot for streaming jobs

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously, all state tables shared a same global committed epoch and safe epoch. To introduce partial checkpoint, each streaming job will have different snapshot (committed and safe epoch). Therefore, in this PR, we introduce `SnapshotGroup`. The state table ids of a streaming job (table fragments) will be in the same group and share a same snapshot, while different streaming jobs can have different snapshot. Though in this PR we will support different snapshots for different streaming jobs, we still maintain that all streaming jobs will have the same snapshot. In the future when we implement and enable partial checkpoint, we can have different snapshots for different streaming jobs.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: proto/hummock.proto [hunk 1]
```diff
@@ -137,6 +137,32 @@ message TableWatermarks {
   bool is_ascending = 2;
 }
 
+message SnapshotGroup {
+  uint32 table_fragments_id = 1;
```
Comment (hzxa21) [inline proto/hummock.proto:141]: nits: do we need plural here? `table_fragment_id`?

**Review by hzxa21 — COMMENTED** [review]
The implementation looks good but I have a general question:
Is `SnapshotGroup` a meta only concept? In other words, do compute/frontent/compactor nodes need to be aware of this concept? Given that we put it in `HummockVersionDelta` and all nodes uses `HummockVersionDelta` to update their local version, it implies that `SnapshotGroup` is "leaked" to all components.

**Comment by wenym1** [comment]
I have changed to maintain snapshot per table. @hzxa21 PTAL

**Review by hzxa21 — APPROVED** [review]
LGTM.

Discussed offline: we can also include compaction group id in `StateTableInfo` and deprecate `member_table_ids`.

**Pull request merged**